internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -111,19 +112,37 @@ func (r *ClusterDeploymentsReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	aciName := clusterDeployment.Spec.ClusterInstallRef.Name
-	err = r.Get(ctx,
+	err := r.Get(ctx,
 		types.NamespacedName{
 			Namespace: clusterDeployment.Namespace,
 			Name:      aciName,
 		},
 		clusterInstall)
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
+			// mark that clusterInstall was already deleted so we skip it if needed.
+			clusterInstallDeleted = true
 			r.Log.WithField("AgentClusterInstall", aciName).Infof("AgentClusterInstall does not exist for ClusterDeployment %s", clusterDeployment.Name)
-			return ctrl.Result{}, nil
+			if clusterDeployment.ObjectMeta.DeletionTimestamp.IsZero() {
+				// we have no agentClusterInstall and clusterDeployment is not being deleted. stop reconciliation.
+				return ctrl.Result{}, nil
+			}
+		} else {
+			r.Log.WithError(err).Errorf("Failed to get AgentClusterInstall %s", aciName)
+			return ctrl.Result{Requeue: true}, err
+		}
+	}
+
+	if !clusterInstallDeleted {
```

Comment:
when can it be true? in line 24 you set this var to be true but then you just return, so maybe this parameters is not needed?

Reply:
`clusterInstallDeleted` is set  to `true` when there is no ACI to fetch (line 122), but will return only if `clusterDeployment` is not being deleted (line 128).

The other return statement in that block is in a case that we failed to fetch ACI for other reasons (line 132)

Reply:
missed that part thanks.